FirebaseRemoteConfigSwift/Sources/PropertyWrapper/RemoteConfigValueObservable.swift

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+import FirebaseRemoteConfig
+
+@available(iOS 14.0, macOS 11.0, macCatalyst 14.0, tvOS 14.0, watchOS 7.0, *)
+internal class RemoteConfigValueObservable<T: Decodable>: ObservableObject {
+  @Published var configValue: T!
```

Comment:
Hi @chliangGoogle and @paulb777, left some more thoughts below.

> On idea to eliminate the force unwraps would be to have type specific property wrappers like RemoteConfigPropertyInt, RemoteConfigPropertyString, RemoteConfigPropertyArray, RemoteConfigPropertyStruct, etc.
> 
> Then they could all use `guard try? ... else set default value` except for RemoteConfigPropertyStruct which would need the property wrapper to take a default value parameter.

This should be able to be done without needed these wrapper types.

We have access to the T type which means we should be able to do something like this:
```swift
 init(key: String, remoteConfig: RemoteConfig) {
    self.key = key
    self.remoteConfig = remoteConfig

    try {
      self.configValue = try remoteConfig[key].decoded()
    } catch {
      print("Decoding of remote config value to given type failed.")
      if T.self is Bool.Type {
        // we are working with a bool and can provide a default value on our own
      } else if T.self is Int.Type {
        // ...
      } else if ... { ... }
      } else {
        // We are working with a custom struct and need to have been given a default to use. 
      }
    }
```

Still though, I still support the idea of enforcing the developer to give default value for _all_ types. The reason being is for two reasons:
- If we don't enforce a default value in the API's signature, the API is guessing really just making an assumption about the "default" behavior of the console-defined remote config schema. For example, would `Bool` default to `false`? What's to say the developer's remote config isn't configured to where they would prefer `true` as the default value. 
- I think of properties as just syntactic sugar. It is my hunch that as the API surface currently is, developers have to have a default value ready if they are using remote config to determine what a UI should look like. They may use the `setDefaults(from:)` API or doing something like:
```swift
struct ContentView: View {
    var body: some View {
        NavigationView {
            VStack {
                // ...
            }
            .onAppear {
                try { 
                    self.welcomeMessage = try remoteConfig["welcome_message"].decoded()
                } catch {
                    self.welcomeMessage = "Welcome!"
                } 
            }
        }
    }
}
```
  Either way, I feel like a default value is going to be needed to fall back on _when using remote config to configure the UI_. 
  Without one, the only other option I see is either hiding the UI element or presenting an error message to the user.

Reply:
I was actually thinking about whether `placeholder` is needed since RC already supports in-app defaults.. so this was helpful to read! 

However I *think* the way RC works is even with no in-app default OR server-side value we still return an "empty" value (0, false, etc), so not entirely sure when we would fallback to the placeholder.. Still agree we should have it though!

Reply:
The fallback placeholder is 4th mechanism after the static. It only reflects on the property wrapper parameter that is not included in the original SDK.